pkg/xds/server.go

```diff
@@ -0,0 +1,455 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"strings"
+	"time"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"istio.io/istio/pilot/pkg/features"
+	istiogrpc "istio.io/istio/pilot/pkg/grpc"
+	istiolog "istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/model"
+	"istio.io/istio/pkg/util/sets"
+)
+
+var log = istiolog.RegisterScope("xdsserver", "xds debugging")
```

Comment:
nit: we might consider using the old `ads` log scope here... normally I wouldn't care much, but these are the most core logs in Istio, so a lot of people have tooling around processing these. It would also be consistent with `delta`, which is still all under the `delta` scope

Reply:
Sure, I can move the scope here.